src/python/pants/engine/isolated_process.py

```diff
@@ -60,6 +64,32 @@ def __new__(
     )
 
 
+class MultiPlatformExecuteProcessRequest(datatype([
+  ('platform_constraints', hashable_string_list),
+  ('execute_process_requests', TypedCollection(Exactly(ExecuteProcessRequest))),
```

Comment:
Why make this a collection? A one-to-one mapping seems more intuitive and simple to me.

Reply:
The underlying representation of the MultiPlatformExecuteProcessRequest on the python side is two lists. I do this so that we can lift the class's fields into rust structs. The TypedCollection represents the values in the request map.